codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/config/ClientConfig.java

```diff
@@ -114,7 +102,7 @@ public String renderDefaults(GenerationContext context) {
     }
 
     public List<ConfigConstraint> getConstraints() {
-        return List.copyOf(this.constraints);
+        return constraints;
```

Comment:
I think in the constructor we still need: `this.constraints = List.copyOf(builder.constraints)`;